src/__tests__/union.test.ts

```diff
@@ -91,6 +153,59 @@ describe('union', () => {
                     done();
                 });
             });
+
+            it('callbacks are only called once', done => {
+                const vol = Volume.fromJSON({
+                    '/foo/bar': 'bar',
+                });
+                const vol2 = Volume.fromJSON({
+                    '/foo/bar': 'baz',
+                });
+                
+                const ufs = new Union() as any;
+                ufs.use(vol as any);
+                ufs.use(vol2 as any);
+
+                const mockCallback = jest.fn();
+                ufs.readFile("/foo/bar", "utf8", () => {
+                    mockCallback();
+                    expect(mockCallback).toBeCalledTimes(1);
+                    done();
+                });
+
+            });
+
+            describe("readdir", () => {
+                it('reads one memfs correctly', () => {
+                    const vol = Volume.fromJSON({
+                        '/foo/bar': 'bar',
+                        '/foo/baz': 'baz',
+                    });
+                    const ufs = new Union();
+                    ufs.use(vol as any);
+                    ufs.readdir("/foo", (err, files) => {
+                        expect(files).toEqual(["bar", "baz"]);
+                    });
+                });
+    
+                it('reads multiple memfs correctly', done => {
+                    const vol = Volume.fromJSON({
+                        '/foo/bar': 'bar',
+                        '/foo/baz': 'baz',
+                    });
+                    const vol2 = Volume.fromJSON({
+                        '/foo/qux': 'baz',
+                    });
+                    
+                    const ufs = new Union();
+                    ufs.use(vol as any);
+                    ufs.use(vol2 as any);
+                    ufs.readdir("/foo", (err, files) => {
+                        expect(files).toEqual(["qux", "bar", "baz"]);
```

Comment:
See my other comment on sorting.

Also, if we do not sort, we cannot guarantee that this array is returned in this order, because this object

```js
{
  '/foo/bar': 'bar',
  '/foo/baz': 'baz',
}
```

is not sorted in JavaScript.